incident.go

```diff
@@ -183,6 +188,37 @@ func (c *Client) ListIncidentsWithContext(ctx context.Context, o ListIncidentsOp
 	return &result, nil
 }
 
+// ListIncidentsPaginated lists existing services processing paginated responses
+func (c *Client) ListIncidentsPaginatedWithContext(ctx context.Context, o ListIncidentsOptions) ([]Incident, error) {
```

Comment:
I don't think this method signature aligns with future plans of the project. The `WithContext` should be removed from the name, as this is a new method and we have no reason to support a non-context APIs for new things. Also, it should include accepting a function that allows filtering of unwanted items.

See https://github.com/PagerDuty/go-pagerduty/issues/295 and https://github.com/PagerDuty/go-pagerduty/issues/296 for more info.